src/components/chart/common-chart-options.js

```diff
@@ -195,6 +195,33 @@ class CommonChartOptions {
             });
         }
     };
+
+    updateLegendForLineSeries = (chart) => {
+        chart.series.forEach((series) => {
```

Comment:
We don't need to loop over the series as well as each legend item here - see the updated version in the other branch.